File: LinkRetrieval.Data/Models/Classes/MatchResult.cs
﻿using LinkRetrieval.Data.Models.Interfaces;
using System.Text.Json.Serialization;

namespace LinkRetrieval.Data.Models.Classes
{
  public class MatchResult : IMatchResult
  {
    public virtual int Id { get; set; }
    public virtual string MatchedText { get; set; }
    public virtual int StartIndex { get; set; }
    public virtual int Length { get; set; }
    public virtual Guid SearchId { get; set; }

    [JsonIgnore]
    public virtual Search Search { get; set; }
  }
}

File: LinkRetrieval.Client/LinkRetrieval/src/app/models/matchResult.ts
export class MatchResult {
  id!: string;
  matchedText!: string;
  startIndex!: number;
  length!: number;
  searchId!: string;

  constructor(id: string, matchedText: string,
    startIndex: number, length: number, searchId: string) {
    this.id = id;
    this.matchedText = matchedText;
    this.startIndex = startIndex;
    this.length = length;
    this.searchId = searchId;
  }
}
